src/lib/services/proposal.ts

```diff
@@ -180,14 +180,21 @@ const zProposalDataResponse = z.object({
         .array(),
       resolved_timestamp: zUtcDate.nullable(),
       submit_time: zUtcDate,
-      total_deposit: zCoin.array(),
+      total_deposit: zCoin.array().nullable(),
       version: z.string(),
       voting_time: zUtcDate.nullable(),
     })
-    .transform<ProposalData>(({ messages, ...val }) => ({
-      ...snakeToCamel(val),
-      messages,
-    }))
+    .transform<ProposalData>(
+      ({ created_tx_hash, proposal_deposits, messages, ...val }) => ({
+        ...snakeToCamel(val),
+        createdTxHash: created_tx_hash ? parseTxHash(created_tx_hash) : null,
+        proposalDeposits: proposal_deposits.map((deposit) => ({
+          ...deposit,
```

Comment:
Let's address each property individually, given that the `deposit` includes a `tx_hash` property.